### DISCUSSION TITLE
remove Server: trillium/0.2.1 header

### DISCUSSION THREAD

**Discussion by prabirshrestha** [trillium-rs/trillium#110] [Q&A]
How do I remove the `Server` header from response? I always consider this to be a security issue since others can know which version of the server we are using and possibly try to exploit known issues.

I have tried the following with no success.

```rust
Init::new(|mut info| async move { info.server_description_mut().clear() }),
```

```rust
|mut conn: Conn| async move {
    conn.headers_mut().remove("Server");
    conn
},
```

Should this header not be added by default to responses?

**Comment by jbr** [comment]
Set the server header to something else if you are concerned about revealing security details. Trillium-http will only set the server header if it has not already been set. There is not currently a way to remove a header, and at the moment that doesn't seem worth the complexity (it would require introducing an empty header value and checking for it in a number of places).

```rust
use trillium::{Conn, KnownHeaderName};
async fn my_server_header(conn: Conn) -> Conn {
  conn.with_header(KnownHeaderName::Server, "my-app")
}
```

---

~~I'll take a look at adding it to the headers before running handlers, but at the moment the solution is as above.~~

The next release of trillium-http will include #113, which will enable `conn.headers_mut().remove(Server)`
